node/runtime/src/lib.rs

```diff
@@ -86,6 +87,28 @@ impl Convert<u128, u128> for CurrencyToVoteHandler {
 	fn convert(x: u128) -> u128 { x * Self::factor() }
 }
 
+pub struct WeightToFeeHandler;
+
+impl Convert<Weight, Balance> for WeightToFeeHandler {
+	fn convert(weight: Weight) -> Balance {
+		// 40/1000000 = 4/100000 = 4/10^5 = 0.00004
+		let variability_fee = Permill::from_parts(40);
+		// 0.00004^2 = 16/10^10 ~= 2/10^9
+		let variability_fee_squared = Perbill::from_parts(2);
+		let potential_weight = <system::Module<Runtime>>::all_extrinsics_weight() + weight;
+		let high_fee: bool =  IDEAL_TRANSACTIONS_WEIGHT <= potential_weight;
+		// workaround because unsigned cannot be negative
+		let diff = if high_fee { potential_weight - IDEAL_TRANSACTIONS_WEIGHT }
+					else { IDEAL_TRANSACTIONS_WEIGHT - potential_weight};
+		let first_term = variability_fee * diff;
+		let second_term = variability_fee_squared * diff * diff / 2;
+		let fee_multiplier = if high_fee { 1 + first_term + second_term }
+								else { 1 + second_term - first_term };
```

Comment:
@Kianenigma Because `variability_fee_squared` is so small, I suspect `fee_multiplier` will always be negative when the total weight is less than the ideal weight (proof: `first_term` >>> `second_term` `=>` `1 - first_term + second_term` < 0 ). **This would imply a problem with the spec itself because if the `fee_multiplier` is negative `=>` `transaction_fee = weight * fee_multiplier` is negative and a negative transaction fee doesn't make any sense. 

If you're wondering why the addition of `1` doesn't ensure it's positive, consider how the `diff` can be very large given that `IDEAL_TRANSACTIONS_WEIGHT` is `1024 * 1024`.